compiler/src/dmd/dsymbolsem.d

```diff
@@ -960,6 +960,9 @@ private extern(C++) final class DsymbolSemanticVisitor : Visitor
                 else if (dsym.type.hasSystemFields())
                     sc.setUnsafePreview(global.params.systemVariables, false, dsym.loc,
                         "`void` initializers for `@system` variables not allowed in safe functions");
+                else if (dsym.type.toBasetype().ty == Tbool)
+                    sc.setUnsafePreview(global.params.systemVariables, false, dsym.loc,
```

Comment:
Can't this just be a deprecation, independent of the system variables preview?

Reply:
I think you're under the impression that system variables don't emit deprecations yet, but they do. The only difference is that it becomes an error already with `-preview=systemVariables`.